ratelimit/doc.go

```diff
@@ -0,0 +1,93 @@
+/*
+Package ratelimit implements rate limiting functionality for the proxy.
+
+It provides per instance based, local, rate limiting. It can be
+configured globally, or based on routes. Rate limiting can be lookuped
```

Comment:
weird term. Does this stand for bucket?

Reply:
It's not the bucket, but it is used to select a bucket. 
For example X-Forwarded-For Lookuper will select the bucket based on the content of this header or the remote ip, in case there is no such header.

I am open to other suggestions.